File: 2003/VB101SamplesAll/VB101SamplesBCL2/BuildingATracingInfrastructure/WebServiceSimulator.vb
Imports System
Imports System.Collections.Generic
Imports System.Text

' This class simulates a web service.
Class WebServiceSimulator

    Public Function GetEmployeeNames(ByVal delayLength As Integer) As String
        ' Retrieve a list of employees.
        ' Simulate a web service responding slowly.
        If (delayLength > 0) Then
            System.Threading.Thread.Sleep(delayLength)
        End If
        ' Return the list if the delay is less than three seconds, otherwise 
        ' return an error string.
        If (delayLength < 4000) Then
            ' Access a database for the information.
            Return ("Nancy Davolio, Andrew Fuller, Janet Leverling, " + "Margaret Peacock, Steven Buchanan")
        Else
            Return "Error"
        End If
    End Function
End Class

File: 2003/CS101SamplesAll/CS101SamplesBCL2/BuildingATracingInfrastructure/WebServiceSimulator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BuildingATracingInfrastructure {

    // This class simulates a web service.
    class WebServiceSimulator {

        public string GetEmployeeNames(int delayLength) {
            // Retrieve a list of employees.
            // Simulate a web service responding slowly.
            if (delayLength > 0) {
                System.Threading.Thread.Sleep(delayLength);
            }

            // Return the list if the delay is less than three seconds, otherwise 
            // return an error string.
            if (delayLength < 4000) {
                // Access a database for the information.
                return "Nancy Davolio, Andrew Fuller, Janet Leverling, " +
                    "Margaret Peacock, Steven Buchanan";

            }
            else {
                return "Error";
            }
        }
    }
}
